### ISSUE TITLE
Running into error while updating stack

### ISSUE THREAD

**Issue by roemhildtg** [swarmpit/swarmpit#485]
---------------------------------------------------
BUG REPORT
---------------------------------------------------

**Description**
I'm not sure this is a bug - I just don't know where I'm going wrong, what I can try to do to fix it. I try to update a service, IE changing the Environment variable or anything really. I see an error, and in the network logs:

Error message (from network logs):
```
{
    "via": [{
        "type": "java.lang.NullPointerException",
        "at": ["clojure.lang.Reflector", "invokeInstanceMethod", "Reflector.java", 97]
    }],
    "trace": [
        ["clojure.lang.Reflector", "invokeInstanceMethod", "Reflector.java", 97],
        ["swarmpit.api$registry_by_url$fn__15985", "invoke", "api.clj", 597],
        ["clojure.core$filter$fn__5878", "invoke", "core.clj", 2817],
        ["clojure.lang.LazySeq", "sval", "LazySeq.java", 42],
        ["clojure.lang.LazySeq", "seq", "LazySeq.java", 51],
        ["clojure.lang.LazySeq", "first", "LazySeq.java", 73],
        ["clojure.lang.RT", "first", "RT.java", 688],
        ["clojure.core$first__5369", "invokeStatic", "core.clj", 55],
        ["clojure.core$first__5369", "invoke", "core.clj", 55],
        ["swarmpit.api$registry_by_url", "invokeStatic", "api.clj", 598],
        ["swarmpit.api$registry_by_url", "invoke", "api.clj", 593],
        ["swarmpit.api$registry_repository_digest", "invokeStatic", "api.clj", 698],
        ["swarmpit.api$registry_repository_digest", "invoke", "api.clj", 694],
        ["swarmpit.api$repository_digest", "invokeStatic", "api.clj", 721],
        ["swarmpit.api$repository_digest", "invoke", "api.clj", 716],
        ["swarmpit.api$standardize_service", "invokeStatic", "api.clj", 922],
        ["swarmpit.api$standardize_service", "invoke", "api.clj", 914],
        ["swarmpit.api$update_service", "invokeStatic", "api.clj", 951],
        ["swarmpit.api$update_service", "invoke", "api.clj", 949],
        ["swarmpit.handler$service_update", "invokeStatic", "handler.clj", 215],
        ["swarmpit.handler$service_update", "invoke", "handler.clj", 211],
        ["reitit.ring.coercion$fn__25661$fn__25663$fn__25664", "invoke", "coercion.cljc", 40],
        ["reitit.ring.middleware.exception$wrap$fn__26541$fn__26542", "invoke", "exception.clj", 49],
        ["ring.middleware.params$wrap_params$fn__22823", "invoke", "params.clj", 67],
        ["buddy.auth.accessrules$wrap_access_rules$fn__17523", "invoke", "accessrules.clj", 347],
        ["swarmpit.authentication$authentication_middleware$fn__18753", "invoke", "authentication.clj", 32],
        ["muuntaja.middleware$wrap_format$fn__26465", "invoke", "middleware.clj", 73],
        ["reitit.ring$ring_handler$fn__23397", "invoke", "ring.cljc", 292],
        ["clojure.lang.AFn", "applyToHelper", "AFn.java", 154],
        ["clojure.lang.AFn", "applyTo", "AFn.java", 144],
        ["clojure.lang.AFunction$1", "doInvoke", "AFunction.java", 31],
        ["clojure.lang.RestFn", "invoke", "RestFn.java", 408],
        ["org.httpkit.server.HttpHandler", "run", "RingHandler.java", 91],
        ["java.util.concurrent.Executors$RunnableAdapter", "call", "Executors.java", 511],
        ["java.util.concurrent.FutureTask", "run", "FutureTask.java", 266],
        ["java.util.concurrent.ThreadPoolExecutor", "runWorker", "ThreadPoolExecutor.java", 1149],
        ["java.util.concurrent.ThreadPoolExecutor$Worker", "run", "ThreadPoolExecutor.java", 624],
        ["java.lang.Thread", "run", "Thread.java", 748]
    ]
}
```

Additionally, when modifying the stack yml in swarmpit I get this network error:
```
{
    "via": [{
        "type": "java.net.MalformedURLException",
        "at": ["java.net.URL", "<init>", "URL.java", 627]
    }, {
        "type": "java.lang.NullPointerException"
    }],
    "trace": []
}
```

**Additional information (e.g. docker version, cluster setup,...):**

Docker Swarm:
EC2 cluster - 
 1 master 4GB RAM 2 CPU 
 2 workers 2GB RAM 2 CPU each

```
Client:
 Version:           18.09.9-ce
 API version:       1.39
 Go version:        go1.10.3
 Git commit:        039a7df
 Built:             Fri Nov  1 19:26:49 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.09.9-ce
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       039a7df
  Built:            Fri Nov  1 19:28:24 2019
  OS/Arch:          linux/amd64
  Experimental:     false
```

**Comment by roemhildtg** [comment]
I think I'm going to close this - since I restarted with a clean slate and got it all working. 

```
docker stack rm swarmpit
docker volume rm swarmpit_data...
```

**Comment by lumir-mrkva** [comment]
@roemhildtg can you share url of images that you deployed in that stack, also have you set any custom registries?

**Comment by roemhildtg** [comment]
Yep - I did have a custom registry added. I think something happened when the following happened:

1. Used swarmpit installer to set up swarmpit
2. Removed stack manually - I wanted to set up docker volumes for persistant storage. 
3. Used docker-compose to deploy stack `docker stack deploy -c docker-compose.yml` - this installed using :latest images
4. Changed docker-compose versions to 1.8 and republished.

One guess is it had to do with downgrading the versions, but the database was on the latest version maybe?
